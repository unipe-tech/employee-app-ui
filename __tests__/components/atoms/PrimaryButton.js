import { render, fireEvent, waitFor } from "@testing-library/react-native";
import { describe, expect, test } from "@jest/globals";
import PrimaryButton from "../../../components/atoms/PrimaryButton";

describe("PrimaryButton", () => {
  test("renders UI correctly", () => {
    const tree = render(<PrimaryButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
