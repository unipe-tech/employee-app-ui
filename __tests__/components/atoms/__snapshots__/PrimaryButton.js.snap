// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryButton renders UI correctly 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#2CB77C",
      "borderBottomEndRadius": 4,
      "borderBottomStartRadius": 4,
      "borderTopEndRadius": 4,
      "borderTopStartRadius": 4,
      "height": 50,
      "justifyContent": "center",
      "marginTop": 29.605263157894736,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 1.41,
    }
  }
  type="solid"
>
  <View
    style={
      Array [
        Object {
          "borderBottomEndRadius": 4,
          "borderBottomStartRadius": 4,
          "borderTopEndRadius": 4,
          "borderTopStartRadius": 4,
          "overflow": "hidden",
        },
        Object {
          "width": "100%",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onLayout={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 36,
            "justifyContent": "center",
            "minWidth": 64,
            "paddingEnd": 16,
            "paddingStart": 16,
          },
          Object {
            "height": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "transition": "background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
